Chessington.GameEngine/Pieces/King.cs

```diff
@@ -10,7 +10,27 @@ public King(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+            
+            int[] row = new int[] { 1, 1, 1, 0, 0, -1, -1, -1};
+            int[] col = new int[] { -1, 0, 1, -1, 1, -1, 0, 1};
+            
+            for (int i = 0; i < 8; i++)
+            {
+                if(board.FindPiece(this).Row + row[i] >= 0
+                   && board.FindPiece(this).Row + row[i] < 8 
+                   && board.FindPiece(this).Col + col[i] >= 0 
+                   && board.FindPiece(this).Col + col[i] < 8 
```

Comment:
I would make a board.IsSquareInBounds(Square) method, to move this logic (and hopefully make it prettier with several if cases). I imagine this could come in handy in a lot of places